vcd/resource_vcd_network_direct.go

```diff
@@ -95,10 +112,103 @@ func resourceVcdNetworkDirectCreate(d *schema.ResourceData, meta interface{}) er
 
 	err = vdc.CreateOrgVDCNetworkWait(orgVDCNetwork)
 	if err != nil {
-		return fmt.Errorf("error: %#v", err)
+		return fmt.Errorf("error: %s", err)
+	}
+
+	network, err := vdc.GetOrgVdcNetworkByName(networkName, true)
+	if err != nil {
+		return fmt.Errorf("error retrieving network %s after creation", networkName)
+	}
+	d.SetId(network.OrgVDCNetwork.ID)
+	return resourceVcdNetworkDirectRead(d, meta)
+}
+
+func resourceVcdNetworkDirectRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf("[network direct read] "+errorRetrievingOrgAndVdc, err)
+	}
+
+	identifier := d.Id()
+
+	if identifier == "" {
```

Comment:
What are cases when id is empty?

Reply:
When the read function is called from the data source

Reply:
Got it. you reusing read function in that way